mailpoet/assets/js/src/segments/dynamic/templates/components/template_list_item.tsx

```diff
@@ -23,12 +24,21 @@ export function TemplateListItem({
 }: TemplateListItemProps): JSX.Element {
   const { createFromTemplate } = useDispatch(storeName);
 
+  const handleSelectingTemplate = (event): void => {
+    event.preventDefault();
+    createFromTemplate(template);
+    MailPoet.trackEvent('Segments > Template selected', {
```

Comment:
Does this work well? The function `createFromTemplate` contains `window.location.href = ...`, which could maybe cut off the tracking with a redirect? I'm not sure, though, but maybe it's safer to call it before, or inside the function?

Reply:
@rodrigoprimo , @JanJakes `window.location.href` might be an issue. Mixpanel offers a callback for solving this.
It is used for example on homepage when tracking before redirecting to posts see https://github.com/mailpoet/mailpoet/blob/a132e6175b645507466c50e9b1f534f9704b9d7c/mailpoet/assets/js/src/homepage/tracking.ts#L4-L13

Reply:
Good catch, I missed this. I added a new commit moving the call to `trackEvent()` to `createFromTemplate()` and using the callback that Rosta mentioned to redirect the user to the segment edit page. Could you please check this PR again?

I had to force push it after rebasing with trunk to fix an unrelated failing integration test.